src/google/signer.rs

```diff
@@ -213,6 +217,40 @@ impl Signer {
 
         Err(anyhow!("token not found"))
     }
+
+    fn canonicalize(
+        &self,
+        req: &impl SignableRequest,
+        method: SigningMethod,
+        cred: &Credential,
+    ) -> Result<CanonicalRequest> {
+        let mut creq = CanonicalRequest::new(
+            req,
+            method,
+            None,
+            SigningAlgorithm::Goog4Rsa,
+            "europe-west3".to_string(),
+            "storage".to_string(),
+        )?;
+        creq.build_headers(cred)?;
+        creq.build_query(cred)?;
+
+        debug!("calculated canonical request: {creq}");
+        Ok(creq)
+    }
+
+    pub fn sign_v4(&self, req: &mut impl SignableRequest) -> Result<()> {
```

Comment:
`sign_v4` is the implementation detail that I don't want to expose to users. Please keep the same API with other signers like `sign_query`.

By the way, we don't need an `expire`?

Reply:
And how do you want to choose between the previous way and v4? We can decide HMAC vs RSA based on credential.

Reply:
> And how do you want to choose between the previous way and v4? We can decide HMAC vs RSA based on credential.

No need to choose at this point. We use bearer token for `sign` and aws v4 for `sign_query`. Other features like gcs's ras based key can be added in the future if we have a request.

Reply:
Hmm, I do think that we "request" this feature and would like to use it. But we can start with only AWS HMAC if you prefer.

Reply:
Oh, sorry. I must miss your feature request. The only feature I remember for is we want OpenDAL to support gcs query sign. 

And I assume that you don't care about the detailed implementation that reqsign use underhood. **Please correct me if I understand incorrectly**. The following design is based on my previous assumption.

So the best option to me is:

- Use AwsSigner to do query sign for gcs. (Please let know if this way doesn't work)
- No other changes (like adding `SigningMethod` or `SigningFlavor`, I don't want to reuse code in this way)